lib/dns/proof.js

```diff
@@ -5,11 +5,29 @@ class Proof {
     this.sig = sig;
     this.sigwire = sigwire;
     this.rrdata = rrdata;
+    this.sigwiredata = Buffer.concat([sigwire, rrdata]);
+  }
+
+  toConcat(rrdata) {
+    let sigwiredatalength = this.sigwiredata.toString('hex').length / 2;
+    let siglength = this.sig.toString('hex').length / 2;
+    let prefixlengh = 4;
+    let buf = Buffer.alloc(sigwiredatalength + siglength + prefixlengh);
+    buf.writeUInt16BE(sigwiredatalength, 0);
+    buf.write(this.sigwiredata.toString('hex'), 2, sigwiredatalength, 'hex');
```

Comment:
It's also not necessary to convert to hex, then back again here - just append the data directly.